dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java

```diff
@@ -0,0 +1,463 @@
+                                                                                                                 // not
+                                                                                                                 // remotely
+                                                                                                                 // initiated
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
+            LoggingUtil.SESSION_LOGGER_CORE.exiting(methodClassName, "invalidateAll");
+        }
+    }
+
+    /*
+     * invalidateAll()
+     * 
+     * @see com.ibm.wsspi.servlet.session.IBMSessionExt#invalidateAll()
+     * Calls invalidateAll(false)
+     */
+    public void invalidateAll() throws SecurityException {
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
+            LoggingUtil.SESSION_LOGGER_CORE.logp(Level.FINE, methodClassName, "invalidateAll", "will default to invalidateAll(false)");
+        }
+        invalidateAll(false);
+    }
+
+    /*
+     * For Session Security Integration feature.
+     * Gets the user name associated with the Session. The user name could be
+     * anonymous (the default that indicates an unregistered visitor), or it could
+     * have been set when it was authenticated by the HTTP server or by an
+     * application.
+     */
+
+    public String getUserName() {
+        return getISession().getUserName();
+    }
+
+    /*
+     * For Session Security Integration feature.
+     * Sets the user name associated with the Session.
+     */
+    public void setUser(String value) {
+        getISession().setUserName(value);
+    }
+
+    /*
+     * For Manual Update, app calls sync() to persist session
+     * 
+     * @see com.ibm.websphere.servlet.session.IBMSession#sync()
+     */
+    public void sync() {
+        getISession().flush();
+    }
+
+    /*
+     * @see javax.servlet.http.HttpSession#getAttribute(java.lang.String)
+     */
+    public Object getAttribute(String pName) {
+        crossoverCheck("getAttribute");
+        return getSessionValue(pName, false);
+    }
+
+    /*
+     * @see javax.servlet.http.HttpSession#getValue(java.lang.String)
+     */
+    public Object getValue(String pName) {
+        return getSessionValue(pName, false);
+    }
+
+    /*
+     * @see javax.servlet.http.HttpSession#getAttributeNames()
+     */
+    
+    /*
+    public Enumeration getAttributeNames() {
+        Enumeration attrNameEnum;
+        if (!_hasSecurityInfo) {
+            attrNameEnum = super.getAttributeNames();
+        } else {
+            String[] attrNames = getValueNames();
+            attrNameEnum = new ArrayEnumeration(attrNames);
+        }
+        return attrNameEnum;
+    }
+    */
+    
+    /*
+     * @see javax.servlet.http.HttpSession#getValueNames()
+     */
+    public String[] getValueNames() {
```

Comment:
I am surprised you are leaving getValue, getValueNames, putValue, and removeValue on the abstract version since it is removed in 6.0.

Reply:
Clean up those removed methods